jolt-core/src/poly/commitment/bmmtv/afgho.rs

```diff
@@ -0,0 +1,90 @@
+//! Afgho commitment
+//! 
+//! This module implements an inner pairing commitment implementation
+
+use std::marker::PhantomData;
+
+use ark_ec::{
+    pairing::{Pairing, PairingOutput},
+    CurveGroup,
+};
+use ark_std::rand::Rng;
+
+use super::inner_products::PairingInnerProduct;
+
+pub type Error = anyhow::Error;
+
+/// Helpers for generator commitment keys used by Pedersen and AFGHO16
+pub fn random_generators<R: Rng, G: CurveGroup>(rng: &mut R, num: usize) -> Vec<G> {
+    (0..num).map(|_| G::rand(rng)).collect()
+}
+
+/// Afgho commitment is simply an inner pairing product commitment
+#[derive(Clone)]
+pub struct AfghoCommitment<P: Pairing> {
+    _pair: PhantomData<P>,
+}
```

Comment:
Might as well put the pairing output in this struct imo
```suggestion
pub struct AfghoCommitment<P: Pairing> {
    g_t: PairingOutput<P>,
}
```